package ExceptionHandling;

import java.io.IOException;

public class ExceptionHandlingDemo {

	public static void main(String[] args) {
		
		try {
			System.out.println("Vor");
			throwIOExeption();
			System.out.println("Nach  der IOEx");
		} catch (IOException e) {
			// TODO Auto-generated catch block
			System.out.println("Vor dem catch der IOEx");
			e.printStackTrace();
			System.out.println("Nach dem catch der IOEx");
		}
		
		throwIllegalArgumentException();
		
//		throwTrhowable();
//		throwAssertionError();
		
	}

	private static void throwIOExeption() throws IOException {
		throw new IOException("My message here");	
	}
	
	private static void throwIllegalArgumentException() throws IllegalArgumentException{
		throw new IllegalArgumentException("Illegal Argument Message here");
	}
	
}
